### PR TITLE
React UI for Creating User

### LINKED ISSUE #190: React for Create User
Create React page for creating a user.

Visit https://nscstrdevusw2thucommon.z5.web.core.windows.net/ to see the form.

### PR DESCRIPTION
**Pull request by jleeboyd**
### Description
Resolves #190 

This PR creates the react page UI and POST request for creating a user in the DB.

This UI feature will be deployed to Azure storage on merge.  For testing locally you will be checking the endpoint https://nsc-fun-dev-usw2-thursday.azurewebsites.net/api/users/ for your newly created user.

https://nscstrdevusw2thucommon.z5.web.core.windows.net/ for the actual site.

**Testing steps are below the screenshots.**

### Screenshots:
**Homepage**
Screen Shot 2021-02-25 at 3 38 46 PM

**Create User Form**
Screen Shot 2021-02-25 at 3 38 57 PM

**Form Validation**
Screen Shot 2021-02-25 at 3 39 04 PM
### To Test:

**User Created**
Screen Shot 2021-02-25 at 3 39 46 PM

**Website Results Showing Creation**
Screen Shot 2021-02-25 at 3 40 06 PM

**Line to Change**
Screen Shot 2021-02-25 at 4 02 22 PM

**Pre-testing steps (depending on if you have initialized the react app previously)**
a. Navigate to the app directory $ cd ui/react-ui
b. Install dependencies $ npm install
c. Start the app $ npm start

**Testing:**
1. Upon starting the app you should see homepage in the screenshot above.
2. Click the blue Create User button at the bottom of the page.
3. You should now be at /create and it should look like the form screenshot above.
4. Try to submit the form with blank input entries, you should see form validation indicating incorrect fields
5. Now fill out the form correctly and click 'Create', you should see the fields reset and an alert notifying you the creation was successful. **Note: the form will reset after successful user creation.**
6. Navigate to the endpoint, [thursday class website](https://nsc-fun-dev-usw2-thursday.azurewebsites.net/api/users/) and make sure your user was successfully created. They should be located at the very bottom of the list. **Note: if you see a blank page the endpoint may be down temporarily.**
7. Now go into Create.js line 49, and change the route to an incorrect address so you can test error handling for the form, ex. remove the /api/
8. Save your file changes in your editor, reload the webpage, and try to correctly resubmit the form using valid entries. You should see an alert telling you there was an error. **Note: the form will reset after unsuccessful user creation.**
9. Play around with the form and make sure nothing breaks.

Time Spent
|**DATE**|**ACTIVITY**|**TIME**|
|---|:---|:---|
|2/24|Code initial route + form UI, read up on bootstrap docs, speak with Farhad about POST to the endpoint|2 hours|
|2/25|Finish form with POST, error handling, testing, documenting |2.5 hours|

Total time: 4.5 hr

### CODE REVIEW

File: ui/react-ui/src/components/routes/create/Create.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        } 
+        // Form is good to go
+        else {
+            await createUser(); // Call function to create user
+            form.reset();
+        }
+    }
+
+    const createUser = async () => {
+        const body = {firstName: firstName, lastName: lastName, email: email};
```
Comment (zxkevinxz) [inline ui/react-ui/src/components/routes/create/Create.js]: You can use shorthand here for this object, if the key name and the variable name are the same you can just put:
{ firstName, lastName, email }

**Review by zxkevinxz — APPROVED** [review]
Looks great, I was able to get it running, and nice documentation. I made one comment if you want to change it, more of just a JavaScript tip.

File: ui/react-ui/src/components/routes/create/Create.js (hunk 1 above)
Comment (jleeboyd) [reply ui/react-ui/src/components/routes/create/Create.js]: Thanks for the tip Kevin, I'll be sure to test it out and incorporate the new syntax!

**Review by farhadbahrehmandhenry — APPROVED** [review]
reviewed and approved - one thing that I would add is a back button to your create component. totally up to you though.

**Pull request merged**